### PR TITLE
ZEN-27937 Proper loading of  Cisco UCS devices with zenbatchload

### PR DESCRIPTION
**Pull request by karolzen**
https://jira.zenoss.com/browse/ZEN-27937
https://github.com/zenoss/ZenPacks.zenoss.CiscoUCSCentral/pull/87

### CODE REVIEW

File: Products/ZenModel/BatchDeviceDumper.py [hunk 1]
```diff
@@ -584,8 +599,11 @@ def getLoaderDefinition(self, obj):
             line = "loader='VMware vSphere', loader_arg_keys=['title', 'hostname', 'username', 'password', 'ssl', 'collector']"
             return name, [line]
 
-        elif path == '/zport/dmd/Devices/CiscoUCS':
-            line = "loader='ciscoucs', loader_arg_keys=['host', 'username', 'password', 'useSsl', 'port', 'collector']"
+        elif path.startswith('/zport/dmd/Devices/CiscoUCS'):
+            if "UCS-Central" in path:
+                line = "loader='ciscoucscentral', loader_arg_keys=['host', 'username', 'password', 'useSsl', 'port', 'collector']"
+            else:
+                line = "loader='ciscoucs', loader_arg_keys=['host', 'username', 'password', 'useSsl', 'port', 'collector', 'ucstype']"
```
Comment (jpeacock-zenoss) [inline Products/ZenModel/BatchDeviceDumper.py]: Occurred to me that the repetition here can be reduced:
```
    loaderArgKeys = ['host', 'username', 'password', 'useSsl', 'port', 'collector']
    if "UCS-Central" in path:
        loaderName = "ciscoucscentral"
    else:
        loaderName = "ciscoucs"
        loaderArgKeys.append("ucstype")
    line = "loader='%s', load_arg_keys=[%s]" % (
        loaderName, ','.join("'%s'" % k for k in loaderArgKeys)
    )
```

Comment (karolzen) [inline Products/ZenModel/BatchDeviceDumper.py]: Indeed, good point :+1:

File: Products/ZenModel/BatchDeviceDumper.py [hunk 2]
```diff
@@ -584,8 +599,14 @@ def getLoaderDefinition(self, obj):
...
-        elif path == '/zport/dmd/Devices/CiscoUCS':
-            line = "loader='ciscoucs', loader_arg_keys=['host', 'username', 'password', 'useSsl', 'port', 'collector']"
+        elif path.startswith('/zport/dmd/Devices/CiscoUCS'):
+            loaderArgKeys = ['host', 'username', 'password', 'useSsl', 'port', 'collector']
+            if "UCS-Central" in path:
+                loaderName = "ciscoucscentral"
+            else:
+                loaderName = "ciscoucs"
+                loaderArgKeys.append("ucstype")
+            line = "loader='%s', load_arg_keys=%r" % (loaderName, loaderArgKeys)
```
Comment (jpeacock-zenoss) [inline Products/ZenModel/BatchDeviceDumper.py]: oops.  should be `loader_arg_keys`, not `load_arg_keys`.

Comment (karolzen) [inline Products/ZenModel/BatchDeviceDumper.py]: yes, fixed.

**Pull request merged**